### ISSUE TITLE
Replace documents in array using positional $ operator

### ISSUE THREAD

**Issue by ckarmann** [mongodb/morphia#1129]
With positional $ operator, Mongo allows not only to update but to replace entirely an element of an array. Morphia doesn't seem to support it.

For example if I have a document like this:

```
{
    "_id" : 1,
    "grades" : [ 
        { "value" : 80, "rank" : 12 }, 
        { "value" : 85, "rank" : 5 }
    ]
}
```

Suppose I want to change the first subdocument in the array "grades". The following update works in Morphia:

```
datastore.update(
                datastore.createQuery(Students.class)
                        .filter("_id", 1)
                        .filter("grades.value", 80) ,
                datastore.createUpdateOperations(Students.class)
                        .set("grades.$.value", 82) // change "value" only.
        );
```

But the following should also work:

```
Grade grade = new Grade(82, 10);
datastore.update(
                datastore.createQuery(Students.class)
                        .filter("_id", 1)
                        .filter("grades.value", 80) ,
                datastore.createUpdateOperations(Students.class)
                        .set("grades.$", grade) // replace completely the matching subdocument
        );
```

But it fails with the following error:

```
org.mongodb.morphia.query.ValidationException: The given path is invalid: grades.$
	at org.mongodb.morphia.internal.PathTarget.resolve(PathTarget.java:110)
	at org.mongodb.morphia.internal.PathTarget.getTarget(PathTarget.java:92)
	at org.mongodb.morphia.query.UpdateOpsImpl.add(UpdateOpsImpl.java:279)
	at org.mongodb.morphia.query.UpdateOpsImpl.set(UpdateOpsImpl.java:223)
 ..............
```

For the record, Mongo supports this very well:

```
db.students.update(
   { _id: 1, "grades.value": 80 },
   { $set: { "grades.$" : { value: 82, rank: 10} } }
)
```

**Comment by evanchooly** [comment]
Looks like simultaneously hit #1127 along with @ckarmann.  I'll take a look at his PR and then your test case but I have a feeling they're the same issue.

**Comment by evanchooly** [comment]
confirmed.  fixed by #1127
